lms/services/organization_usage_report.py

```diff
@@ -33,16 +34,61 @@ class UsageReportRow:
 class OrganizationUsageReportService:
     def __init__(
         self,
-        db_session: Session,
+        db: Session,
         h_api: HAPI,
         organization_service: OrganizationService,
     ):
-        self._db_session = db_session
+        self._db = db
         self._organization_service = organization_service
         self._h_api = h_api
 
+    def get(
+        self, organization, tag, since: date, until: date
+    ) -> OrganizationUsageReport | None:
+        key = OrganizationUsageReport.generate_key(organization, tag, since, until)
+        return self.get_by_key(key)
+
+    @staticmethod
+    def monthly_report_dates(
```

Comment:
Doing all the dates math here

This migh involve to:

- Not call `today()` here but pass the value
- Don't assume `monthly` but pass a period as a value

We don't yet need those so I opted for writing this and a unit test with cases covering different edge cases.